### ISSUE TITLE
Trouble getting new oauth to work properly.

### ISSUE THREAD

**Issue by 4Dolio** [tchellomello/python-ring-doorbell#186]
I am following from https://github.com/tchellomello/python-ring-doorbell/issues/182

I am still having trouble, after switching from the main page example(https://github.com/tchellomello/python-ring-doorbell), and the older documentation example(https://python-ring-doorbell.readthedocs.io/), and now looking at the test.py example mentioned in the above issue.  When I use the test.py I get this error.

> Username: me@internet.net
> Traceback (most recent call last):
>   File "./ring.basic.py", line 147, in <module>
>     main()
>   File "./ring.basic.py", line 132, in main
>     username = input("Username: ")
>   File "<string>", line 1
>     me@internet.net
>        ^
> SyntaxError: invalid syntax

OR this:
> Username: asdf
> Traceback (most recent call last):
>   File "./ring.basic.py", line 147, in <module>
>     main()
>   File "./ring.basic.py", line 132, in main
>     username = input("Username: ")
>   File "<string>", line 1, in <module>
> NameError: name 'asdf' is not defined

Previously I had my own wrapper .py script that would just scrape(download) all my clips if I did not already have the file, for archival sake, nothing more.  That script takes my own set of arguments, including my username(email) and password, and I do not have 2fa enabled for my account, yet.  Commenting out those two input lines asking for username and password, that it seems I can not input, does not help, results in 
> Traceback (most recent call last):
>   File "./ring.basic.py", line 147, in <module>
>     main()
>   File "./ring.basic.py", line 136, in main
>     auth = Auth(None, token_updated)
>   File "/usr/local/lib/python2.7/dist-packages/ring_doorbell/auth.py", line 21, in __init__
>     client=LegacyApplicationClient(client_id=OAuth.CLIENT_ID), token=token
>   File "/home/foo/.local/lib/python2.7/site-packages/requests_oauthlib/oauth2_session.py", line > 79, in __init__
>     self.token = token or {}
>   File "/home/foo/.local/lib/python2.7/site-packages/requests_oauthlib/oauth2_session.py", line > 129, in token
>     self._client.populate_token_attributes(value)
>   File "/home/foo/.local/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/clients/base.py", line 492, in populate_token_attributes
>     if 'access_token' in response:
> TypeError: argument of type 'function' is not iterable

I also attempted to restore the past `.ring_doorbell-session.cache` file to see how that behaved and got this:
> Traceback (most recent call last):
>   File "./ring.basic.py", line 148, in <module>
>     main()
>   File "./ring.basic.py", line 129, in main
>     json.loads(cache_file.read_text()),
> AttributeError: 'PosixPath' object has no attribute 'read_text'

Also fixed this typo:
>    print(f"Hello {ring.session['profile']['first_name']}")
>                                                                      ^ (closing quote, ditch the extra "f")
> SyntaxError: invalid syntax

Found if I double-quote my username and password responses, then it proceeds until:
> Traceback (most recent call last):
>   File "./test.py", line 46, in <module>
>     main()
>   File "./test.py", line 31, in main
>     auth = Auth(None, token_updated)
>   File "/usr/local/lib/python2.7/dist-packages/ring_doorbell/auth.py", line 21, in __init__
    client=LegacyApplicationClient(client_id=OAuth.CLIENT_ID), token=token
>   File "/home/foo/.local/lib/python2.7/site-packages/requests_oauthlib/oauth2_session.py", line 79, in __init__
>     self.token = token or {}
>   File "/home/foo/.local/lib/python2.7/site-packages/requests_oauthlib/oauth2_session.py", line 129, in token
>     self._client.populate_token_attributes(value)
>   File "/home/foo/.local/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/clients/base.py", line 492, in populate_token_attributes
>     if 'access_token' in response:
> TypeError: argument of type 'function' is not iterable

**Comment by steve-gombos** [comment]
I'll try to address what I can.  Looks like you are running with python 2.7.  I think a discussion needs to be made as to whether this repo should still support python 2.7 as it went end of life on Jan 1.  Most of the issues I think you are running into are because of that.

The old cache file will not work, as the existing cache file code was removed.  There is a token_updater that is passed into the Auth class, and this method is responsible for accepting the token and you can do whatever with it to save it off and reuse.  The example has this covered by writing it to a new cache file.

**Comment by 4Dolio** [comment]
Don't think I am doing any other python on this system, I can upgrade if that will help. Or I can hold back if you want to try and keep 2.7 compatibility?  I will bring up a newer environment and take another pass at it.  Thanks for the tip.

**Comment by 4Dolio** [comment]
I am now trying with python3.5, fixed a few of my own errors, but am still getting:
> Traceback (most recent call last):
>   File "./ring.basic.py", line 154, in <module>
>     main()
>   File "./ring.basic.py", line 142, in main
>     auth = Auth(None, token_updated)
>   File "/usr/local/lib/python3.5/dist-packages/ring_doorbell/auth.py", line 21, in __init__
>     client=LegacyApplicationClient(client_id=OAuth.CLIENT_ID), token=token
>   File "/usr/local/lib/python3.5/dist-packages/requests_oauthlib/oauth2_session.py", line 79, in __init__
>     self.token = token or {}
>   File "/usr/local/lib/python3.5/dist-packages/requests_oauthlib/oauth2_session.py", line 129, in token
>     self._client.populate_token_attributes(value)
>   File "/usr/local/lib/python3.5/dist-packages/oauthlib/oauth2/rfc6749/clients/base.py", line 492, in populate_token_attributes
>     if 'access_token' in response:
> TypeError: argument of type 'function' is not iterable

**Comment by steve-gombos** [comment]
This is due to the lack of a query string parameter when declaring the auth class.  Documentation should be updated now.

**Comment by 4Dolio** [comment]
Yippee, after some wrangling of packages to get from python 3.5 to 3.6, I was able to get the test script to work... and then to work all the new bugs out of my own script.  All I am using it for is scraping(downloading) the clips, with the ability to specify some rough timespans, to go fetch older missed clips.  Now catching back up on clips from 2019.12.18 through 12.31 and then on to the January clips, then finally back into the "just keepup" mode.  Is there some frequency I should stay below? I think I have been polling either every 15minutes or 60 minutes.  I have one camera which has a lot of activity on it but most do not.  I do verify that I do not already have the clip, and skip the download if I do already have the file.  I have worked out a custom file name that I like, which is YYYYMMDD.HHMMSS-Camera_Name-ClipID.mp4.  Anyway, cheers, and thanks for the API.
